fairseq/sequence_generator.py

```diff
@@ -277,6 +284,13 @@ def _generate(
         # number of candidate hypos per step
         cand_size = 2 * beam_size  # 2 x beam size in case half are EOS
 
+        # Initialize constraints, when active
+        self.search.init_negative_constraints(negative_constraints, cand_size)
+        # handle <s>
+        init_cand_indices = torch.ones(bsz, cand_size)*(self.eos if bos_token is None else bos_token)
+        init_cand_beams = torch.arange(cand_size).repeat(bsz, 1).to(src_tokens.device).long()
+        self.search.update_negative_constraints(init_cand_indices, init_cand_beams)
```

Comment:
This can also be initialized lazily on the first call to `LexicallyConstrainedBeamSearch::step`

Reply:
Done.